app/main.py

```diff
@@ -1,2 +1,70 @@
+from collections.abc import Hashable
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+        self.next = None
```

Comment:
You don't really need to store `next` Node. This would make sense in a linked list, but here you can simply check if next index is free or not in case of collision